File: book/문자열뒤집기.py
nums = input()

count0 = 0
count1 = 0

now = nums[0]
if now == '1':
    count1 += 1
else:
    count0 += 1

for num in nums[1:]:
    if now != num:
        if num == '0':
            count0 += 1
        else:
            count1 += 1
    now = num
print(min(count0, count1))

File: book/문자열뒤집기.java
package main;

import java.util.*;


public class Main {


    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String s = sc.next();

        int count0 = 0;
        int count1 = 0;

        if (s.charAt(0) == '1') count0 += 1;
        else count1 += 1;

        for (int i = 0; i < s.length() - 1; i++) {
            if (s.charAt(i) != s.charAt(i + 1)) {
                if (s.charAt(i + 1) == '1') count0 += 1;
                else count1 += 1;
            }
        }

        System.out.println(Math.min(count0, count1));

    }
}

/*
0001100
 */
